src/libsyntax/parse/parser.rs

```diff
@@ -6203,8 +6204,10 @@ impl<'a> Parser<'a> {
                 // This mod is in an external file. Let's go get it!
                 let ModulePathSuccess { path, directory_ownership, warn } =
                     self.submod_path(id, &outer_attrs, id_span)?;
-                let (module, mut attrs) =
+                let (mut module, mut attrs) =
                     self.eval_src_mod(path, directory_ownership, id.to_string(), id_span)?;
+                // Record that we fetched the mod from an external file
+                module.inline = false;
```

Comment:
This is a relatively minor point, but: You might as well move this line into the body of `fn eval_src_mod`, right? We might as well have it fully encapsulate setting up all the state related to reading a file as a module, rather than leaving this one step for its caller...

Reply:
Yes, I was afraid maybe someone would call `eval_src_mod` from somewhere else, but later I discovered this is the only place it is called.